osquery/events/linux/http_event_publisher.h

```diff
@@ -0,0 +1,133 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <pcap.h>
+#include <osquery/events.h>
+
+namespace osquery {
+
+/**
+ * @brief Subscriptioning details for HTTPLookupEventPublisher events.
+ *
+ */
+struct HTTPLookupSubscriptionContext : public SubscriptionContext {};
+
+struct HTTPLookupEventContext : public EventContext {
```

Comment:
A single event context should collect multiple event items in order to optimize database writes/reads (thanks to write batching). Without this change, a high number of events/second will eventually cause CPU/memory spikes and lock/break the database